# Pancreatic cancer prognostic markers 
In this example notebook, we will analyse pancreatic cancer prognistic markers and link them to biological pathways to identify pathways linked to survival prognosis. 

### Step 1: Load prognostic markers
We will start by loading known favorable prognsis markers from the TCGA project.

In [17]:
data <- read.table(file="data/panc_can_favorable_markers.tsv", sep="\t", header=T)
# extract pancreatic relevant columns
data.filt <- data[,c(1,3,5,17,21,25,76,112,279)]
dim(data.filt)
head(data.filt)

[1] 857   9

Gene   Ensembl         Uniprot RNA.tissue.specificity
1 CAMTA2 ENSG00000108509 O94983  Tissue enhanced       
2 TSPYL2 ENSG00000184205 Q9H2G4  Tissue enhanced       
3 ANAPC2 ENSG00000176248 Q9UJX6  Low tissue specificity
4 DEF8   ENSG00000140995 Q6ZN54  Low tissue specificity
5 USP20  ENSG00000136878 Q9Y2K6  Low tissue specificity
6 MICAL1 ENSG00000135596 Q8TDZ2  Low tissue specificity
  RNA.single.cell.type.specificity RNA.cancer.specificity
1 Low cell type specificity        Low cancer specificity
2 Low cell type specificity        Low cancer specificity
3 Low cell type specificity        Low cancer specificity
4 Low cell type specificity        Low cancer specificity
5 Low cell type specificity        Low cancer specificity
6 Low cell type specificity        Low cancer specificity
  Pathology.prognostics...Pancreatic.cancer Tissue.RNA...pancreas..NX.
1 prognostic favorable (6.00e-8)            11.7                      
2 prognostic favorable (6.74e-8)            11.7                      
3 prognostic favorable (1.07e-7)            23.7                      
4 prognostic favorable (1.35e-7)            22.5                      
5 prognostic favorable (1.54e-7)            12.9                      
6 prognostic favorable (2.63e-7)             4.6                      
  Single.Cell.Type.RNA...Pancreatic.endocrine.cells..NX.
1  9.1                                                  
2  4.4                                                  
3  6.9                                                  
4 83.0                                                  
5  2.3                                                  
6 10.6

The data contains 857 genes with a favorable prognosis for pancreatic cancer. 

### Step 2: Link genes to pathways


In [19]:
library(rWikiPathways)
wp.hs.gmt <- rWikiPathways::downloadPathwayArchive(organism="Homo sapiens", format = "gmt")
wp2gene <- readPathwayGMT(wp.hs.gmt)
wpid2gene <- wp2gene %>% dplyr::select(wpid,gene) #TERM2GENE
wpid2name <- wp2gene %>% dplyr::select(wpid,name) #TERM2NAME

ewp.up <- clusterProfiler::enricher(
        data.filt$Ensembl,
        pAdjustMethod = "fdr",
        pvalueCutoff = 0.05,
        TERM2GENE = wpid2gene,
        TERM2NAME = wpid2name)

head(ewp.up)

ERROR: Error in library(rWikiPathways): there is no package called ‘rWikiPathways’
